src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -553,13 +553,22 @@ func newPromTSIter(timeseries []prompb.TimeSeries, tagOpts models.TagOptions) (*
 		datapoints       = make([]ts.Datapoints, 0, len(timeseries))
 		seriesAttributes = make([]ts.SeriesAttributes, 0, len(timeseries))
 	)
+	graphiteTagOpts := tagOpts
+	graphiteTagOpts.SetIDSchemeType(models.TypeGraphite)
```

Comment:
This needs to be:
```
graphiteTagOpts = graphiteTagOpts.SetIDSchemeType(models.TypeGraphite)
```

Since options are immutable and you need to create a modified version of the options.

Or more readable:
```
graphiteTagOpts := tagOpts.SetIDSchemeType(models.TypeGraphite)
```